packages/utils/src/consolidated/bulk-exists-consolidates.ts

```diff
@@ -0,0 +1,133 @@
+import * as dotenv from "dotenv";
+dotenv.config();
+// keep that ^ on top
+import { createConsolidatedDataFilePath } from "@metriport/core/domain/consolidated/filename";
+import { S3Utils } from "@metriport/core/external/aws/s3";
+import { executeAsynchronously } from "@metriport/core/util/concurrency";
+import { getFileContents } from "@metriport/core/util/fs";
+import { out } from "@metriport/core/util/log";
+import { errorToString, getEnvVarOrFail, sleep } from "@metriport/shared";
+import { buildDayjs } from "@metriport/shared/common/date";
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+import fs from "fs";
+import { elapsedTimeAsStr } from "../shared/duration";
+import { initFile } from "../shared/file";
+import { buildGetDirPathInside, initRunsFolder } from "../shared/folder";
+import { getCxData } from "../shared/get-cx-data";
+
+dayjs.extend(duration);
+
+/**
+ * This script checks if consolidated data exists for a list of patients.
+ * It saves the results to files in the `runs/exists-consolidated` folder:
+ * - <cx-name>_<timestamp>_exists.txt: Patient IDs for which consolidated bundle exists
+ * - <cx-name>_<timestamp>_missing.txt: Patient IDs for which consolidated bundle does not exist
+ *
+ * Execute this with:
+ * $ ts-node src/consolidated/bulk-exists-consolidates.ts
+ */
+
+// Add patient IDs here to kick off queries for specific patient IDs
+const patientIds: string[] = [];
+// Alternatively, you can provide a file with patient IDs, one per line
+const fileName = "";
+
+const numberOfParallelExecutions = 30;
+
+const cxId = getEnvVarOrFail("CX_ID");
+const medicalDocsBucketName = getEnvVarOrFail("MEDICAL_DOCUMENTS_BUCKET_NAME");
+const region = getEnvVarOrFail("AWS_REGION");
+const s3Utils = new S3Utils(region);
+
+// query stuff
+const confirmationTime = dayjs.duration(10, "seconds");
+
+// output stuff
+const getOutputFileName = buildGetDirPathInside(`exists-consolidated`);
+
+async function main() {
+  initRunsFolder();
+  const { log } = out("");
```

Comment:
Is this intentionally an empty string?

Reply:
yeah, that function can be called in diff ways, we could improve that, but since those are just utils we end up just using like that 😄